dynawo/sources/Common/DYNTrace.cpp

```diff
@@ -137,30 +139,44 @@ bool Trace::isLoggingEnabled() {
   return logging::core::get()->get_logging_enabled();
 }
 
-void Trace::addAppenders(const std::vector<TraceAppender>& appenders) {
+void Trace::addAppenders(std::vector<TraceAppender>& appenders) {
   instance().addAppenders_(appenders);
 }
 
-void Trace::addAppenders_(const std::vector<TraceAppender>& appenders) {
+void Trace::addAppenders_(std::vector<TraceAppender>& appenders) {
   // remove old appenders (console_log)
   Trace::resetCustomAppenders();
 
   logging::attributes::current_thread_id::value_type currentId =
     logging::attributes::current_thread_id().get_value().extract<logging::attributes::current_thread_id::value_type>().get();
 
+  for (TraceAppender& appender : appenders) {
```

Comment:
Not needed and dangerous: the variable log is not persistant (persistang logs are the logs from main caller like dyn-algo or dfl that needs to keep existing after the simulation)

Reply:
please write a generic method similar to the resetCustomPersistantAppender and using sinks instead of persistantSink and use it to retrieve the variables sink when iwe need to erase it